airflow/_vendor/nvd3/__init__.py

```diff
@@ -16,14 +16,14 @@
            'scatterChart', 'discreteBarChart', 'multiBarChart']
 
 
+from . import ipynb
+from .cumulativeLineChart import cumulativeLineChart
```

Comment:
Can you exclude anything under airflow/_vendor from changes please?

Perhaps `skip=airflow/_vendor` in setup.cfg might do the trick? Should do given `skip = build,.tox,venv` is used in some of isort's examples.

Reply:
I have another thought. 

I think one big change to order all imports in one go is not a good idea. We should probably use the same approach as we did for pylint.

Since we can have all of this as pre-commit hooks - we could do filtering in the same way we do for pylint pre-commit hooks/ci_scripts using the very same "pylint_todo.txt" file. 
I think those two are fairly connected and we should use the list to only introduce such changes when we remove files from the pylint_todo.txt. And this way we make sure it will be gradually introduced.
 
Once we add it as pre-commit hook, it will have the property that someone fixing pylint will also have to make sure that imports are sorted. I also think about adding pydocstyle in exactly the same way -> so make sure the style is defined and checked for all files that are not on the pylint_todo.txt. This way reviewers will not have to even check if the documentation style we have is good and follows our agreed conventions. I think this should be automated rather than taking precious time of reviewers :).

What do you think (@KevinYang21 @BasPH @ashb @mik-laj ) ? I can add it quickly while I will be adding pydocstyle (I plan to test it while fixing @mik-laj comments for #5786) and I can add isort in very similar way there.

Reply:
I think that this change should be combined with automatic code formatting. Some automatic formatting tools also sort imports. If we now introduce import sorting, we will limit the choice of a tool that will be compatible, or we will have to re-sort all imports this time in a manner compatible with another tool.

Reply:
I think we should treat pylint + pydoc + import sort differently than whole source file formatting change. 

Black formatting (or whatever we choose) changes whole files. Introducing it will basically rewrite ALL the lines of code we have. It will make it next to impossible to merge in-progress PRs (and in some cases those changes will be very difficult to resolve conflicts in those rebases). 

On the other hand - sorting imports/adding pylint changes/updating documentation is a very small and localised change to a small part of the code - even if it is done in all files.
So changing import orders or adding docs is far less invasive than formatting change (you can still easily merge other changes after you applied them).

That's why I think it's worth to sort imports now separately. Even if formatting tool will change the sort order later, it will be done together with plenty of other changes for those files, so this will be pretty much no-issue if also sorting order for imports change then.

Reply:
This change shouldn't affect _all_ PRs, only ones that new imports.

With the `git blame --ignore-revs-file` option (I thought there was a default way to set that but it's a client only setting) I would prefer changes like this to be a single commit rather than multiple.

Reply:
@ashb - If you refer to import ordering I think import sort is rather non-invasive so I do not think it falls into the 'git blame' case. It is rather localised and does not really affect the "meat" of the source code. But If you refer to reformatting - I agree that re-formatting the code if we decide to do it should be done as single big change and one that we can exclude with --ignore-revs-file.

Reply:
BTW. Some goodies from Chromium team: git hyper-blame command - one that can ignore commits stored in committed .git-blame-ignore-revs  (https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/git-hyper-blame.html)

:D

Reply:
Will exclude unnecessary packages. Given how automatic isort can be esp. with pre-commit, I prefer to roll the change out in one commit.